server/src/main/webapp/WEB-INF/rails/webpack/views/pages/pipeline_activity/pipeline_run_info_widget.tsx

```diff
@@ -77,6 +79,9 @@ export class PipelineRunWidget extends MithrilViewComponent<PipelineRunAttrs> {
             </div>
           </div>;
         })}
+        <CommentWidget comment={pipelineRunInfo.comment}
+                       counterOrLabel={pipelineRunInfo.counterOrLabel()}
+                       addOrUpdateComment={vnode.attrs.addOrUpdateComment}/>
```

Comment:
The page is shown when the user has `view` permissions. But the comment feature is available only when `operate` permission is provided. Should we check that before showing the button to add?

Reply:
Yes, Ideally it should be visible only when the user has `operate` permission. There is no point in showing the button as an API call will always fail.

Reply:
Also, pls check the following scenario:
1. `user1` adds a comment.
2. Another `user2` can view it but cannot edit or remove it.

Reply:
@kritika-singh3  
- Currently, anyone can add/edit the comment on the pipeline instance if it has `operate` permission on the pipeline(same as old SPA).

- Adding comment will fail if the user don't have the `operate` permission on the pipeline